In [1]:
import pandas as pd
import time
import seaborn as sns
import datetime as dt
import numpy as np
import matplotlib.pyplot as plt
import os
from tqdm import tqdm

import pokerDataParser

In [2]:
path = os.path.join('C:\\Users','templ','Documents','GitHub','gambling','texasHoldEm','game_data','Export Holdem Manager 2.0 12302016144830.txt')

t = pokerDataParser.parseGames(path)

test = pokerDataParser.analyzeGame(t[2])

In [3]:
%%timeit
pokerDataParser.analyzeGame(t[2])

18.1 ms ± 216 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)


In [4]:
print('number of individual games in data: {}'.format(len(t)))

number of individual games in data: 42688


Functions above show the individual computation time to parse the dataset and the total number of games, 42688. 

In [5]:
master = pd.DataFrame()

for data in tqdm(t):
    out = pokerDataParser.analyzeGame(data)
    master = master.append(out)

100%|████████████████████████████████████████████████████████████████████████████| 42688/42688 [49:50<00:00, 14.27it/s]


In [6]:
master

betDollar  gameWinner  allIn  fold  \
gameId    playerName   round                                       
808941103 IlxxxlI      hole        4.00           0      0     1   
                       flop        0.00           0      0     0   
                       turn        0.00           0      0     0   
                       river       0.00           0      0     0   
808943744 IlxxxlI      hole        3.00           1      0     0   
...                                 ...         ...    ...   ...   
788414106 dogwipe      river       5.41           1      0     0   
          JerredWilson hole        1.00           0      0     0   
                       flop        1.42           0      0     0   
                       turn        2.77           0      0     0   
                       river       5.41           0      0     0   

                                     cards  bankroll  betPercent  isBigBlind  \
gameId    playerName   round                                                   
808941103 IlxxxlI      hole     [D11, C11]     53.50    0.074766           0   
                       flop             []     53.50    0.000000           0   
                       turn             []     53.50    0.000000           0   
                       river            []     53.50    0.000000           0   
808943744 IlxxxlI      hole      [C9, C14]     49.50    0.060606           0   
...                                    ...       ...         ...         ...   
788414106 dogwipe      river         [D14]     45.31    0.119400           0   
          JerredWilson hole       [C9, S8]     42.67    0.023436           0   
                       flop   [D7, S2, D8]     42.67    0.033279           0   
                       turn          [D13]     42.67    0.064917           0   
                       river         [D14]     42.67    0.126787           0   

                              isSmallBlind  bigBlind  smallBlind  
gameId    playerName   round                                      
808941103 IlxxxlI      hole              0       1.0        0.50  
                       flop              0       1.0        0.50  
                       turn              0       1.0        0.50  
                       river             0       1.0        0.50  
808943744 IlxxxlI      hole              0       1.0        0.50  
...                                    ...       ...         ...  
788414106 dogwipe      river             0       0.5        0.25  
          JerredWilson hole              1       0.5        0.25  
                       flop              1       0.5        0.25  
                       turn              1       0.5        0.25  
                       river             1       0.5        0.25  

[201352 rows x 11 columns]